### PR TITLE
Add filter to activities by event

### LINKED ISSUE #84: User should be able to filter activities by event
**As User**
**I want** to be able to filter activity data by event

### Acceptance Criteria

```gherkin
Scenario: 
Given a User
When he makes activities GraphQL request with `events` argument
Then I should receive activities data only for specified events
```
```ruby
me {
  id
  email
  activities(events: ["USER REGISTERED"]) {
    id
    title
    body
    event
    createdAt
  }
}
```

**Notes:**
N/A

### PR DESCRIPTION
**Pull request by AzatZaynutdinov**
### Summary

User should be able to filter activities by event

### How it works

image

### Test plan

List of steps to manually test introduced functionality:

* Open GraphiQL App
* Make request using schema:
```
query {
  me {
    id
    email
    activities(events: [USER_REGISTERED]){
      id
      event
    }
  }
}
```
* ...

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, ENV variables, services, scripts, etc.

### References
* Resolves #84  Issue

### CODE REVIEW

File: spec/graphql/types/query_type_me_spec.rb [hunk 1]
```diff
@@ -31,5 +32,39 @@
...
+          }
+        GRAPHQL
+      end
+
+      it_behaves_like "graphql request", "gets current_user info" do
+        let(:schema_context) { { current_user: user, token_payload: token_payload.stringify_keys } }
+        let(:fixture_path) { "json/acceptance/graphql/query_type_me_with_activities.json" }
+        let(:prepared_fixture_file) do
+          fixture_file.gsub(
+            /:id|:email|:first_name|:last_name|:activity_id/,
```
Comment (EvgeniyEsaulkov) [inline spec/graphql/types/query_type_me_spec.rb:59]: Could you please hardcode values for `email`, `first_name` and `last_name` in the fixture file?

File: app/graphql/types/user_type.rb [hunk 2]
```diff
@@ -5,11 +5,8 @@ class UserType < Types::BaseObject
     field :first_name, String, null: true
     field :last_name, String, null: true
     field :avatar_url, String, null: true
-    field :activities, [ActivityType], null: true
 
-    def activities
-      Loaders::AssociationLoader.for(User, :activities).load(object)
-    end
+    field :activities, resolver: Resolvers::Activities
```
Comment (EvgeniyEsaulkov) [inline app/graphql/types/user_type.rb:9]: This will include activities for all users, not only current user

Comment (AzatZaynutdinov) [reply app/graphql/types/user_type.rb:9]: fixed this way
https://github.com/fs/rails-base-graphql-api/pull/89/files#diff-e3e54d8f05fa6dbbd16d5ce8453b310cR12

File: spec/query_objects/filtered_activities_query_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    it { is_expected.to match_array([activity_1, activity_2]) }
+
+    context "when param is user_updated" do
+      let(:filter_params) { { events: [:user_updated] } }
+
+      it { is_expected.to match_array([activity_2]) }
+    end
+
+    context "when param is user_registered" do
```
Comment (EvgeniyEsaulkov) [inline spec/query_objects/filtered_activities_query_spec.rb:24]: Are you sure we need tests for each event type?

Comment (AzatZaynutdinov) [reply spec/query_objects/filtered_activities_query_spec.rb:24]: I agree that's not necessary for every type.

File: spec/query_objects/filtered_activities_query_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      let(:filter_params) { { events: [:user_registered] } }
+
+      it { is_expected.to match_array([activity_1]) }
+    end
```
Comment (ArthurZaharov) [inline spec/query_objects/filtered_activities_query_spec.rb:28]: We can add a spec for empty events array. What should we return if client makes next query
```
query {
  activities(events: []) {
    id
    title
    user {
      id
      email
    }
  }
}
```
Should we return empty activities list (as it is now) or we should return all activities?

**Comment by ArthurZaharov** [comment]
@AzatZaynutdinov @EvgeniyEsaulkov What is the status of this PR?

**Comment by AzatZaynutdinov** [comment]
@ArthurZaharov I fixed the comments. Status - waiting for approve

**Pull request merged**